### DISCUSSION TITLE
Mounting a new EPHEMERAL partition?

### DISCUSSION THREAD

**Discussion by mjohnson9** [siderolabs/talos#8183] [Q&A] [answered]
I'd like to install Talos' bootloader on an SD card (actually a pair of mirrored SD cards, part of a Dell Internal SD Module), and have `EPHEMERAL` (and `STATE`, depending on the volume of writes it receives) on a different disk that is faster and more durable. The reason for this is that my server, a Dell T330, cannot boot from NVMe due to BIOS limitations. On other OSes, I solve this the same way: the bootloader goes on the SD card.

Ideally, I could mount the `STATE` and `EPHEMERAL` partitions from one of the node's ZFS pools. This would mean not consuming a valuable NVMe drive that could otherwise be added to the pool.

If I add a `/var` to `extraMounts`, would that reliably solve this? I know Linux handles multiple mounts at the same mount point with the most recently mounted one having priority, but that depends on a number of factors:

1. `extraMounts` always happening after the system's default `EPHEMERAL` mount
2. Nothing taking file handles from the default mount before `extraMounts
3. The ZFS extension's kernel module being available at the time that `extraMount` happens.

This sounds pretty fragile to me, though. I didn't find any history of discussing this here, but: Has anyone else run into and solved something similar?

Another solution that I'm considering is chaining bootloaders, with something like GRUB or rEFInd on the SD cards to load the NVMe driver and start Talos' GRUB. This doesn't have the benefit of being able to use the NVMe drive in my ZFS pool, though.

**Comment by smira** [comment, marked as answer]
This is not supported right now, but might be available in Talos 1.7

**Reply by nogweii** [reply]
Reading the release notes and changelog for 1.7, this does not seem to have landed. Will it be in 1.8? Or perhaps a later version?

**Reply by smira** [reply]
Please follow this issue https://github.com/siderolabs/talos/issues/8367, it got shifted to Talos 1.8
